.github/workflows/release.yml

```diff
@@ -0,0 +1,49 @@
+name: Release
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+concurrency:
+  group: build-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04] # TODO: macos-11
+      fail-fast: false
+    env:
+      GH_TOKEN: ${{ github.token }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: hendrikmuhs/ccache-action@v1
+        if: matrix.os == 'ubuntu-20.04' || matrix.os == 'macos-11'
+
+      - name: Get short git SHA
+        id: sha
+        run: echo "GIT_SHA_SHORT=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Install system dependencies (Ubuntu)
+        run: ./scripts/install_deps_ubuntu.sh
+        if: matrix.os == 'ubuntu-20.04'
+
+      - name: Install system dependencies (Mac)
+        run: ./scripts/install_deps_mac.sh
+        if: matrix.os == 'macos-11'
+
+      - name: Install venv dependencies
+        run: ./bootstrap_venv.sh
+
+      - name: Create and upload release
+        run: |
+          git tag ${{ steps.sha.outputs.GIT_SHA_SHORT }}
```

Comment:
Do you need to tag if its a release?  having git shas as tags seems confusing and possibly ambiguous?  Maybe we it should be the commit number + branch name?

Reply:
We need a tag for a release, yes. I modified to `v.<short_sha>`. Trying to use a convention that's a simple function of the git submodule's sha, to avoid complicating the automation further.